server/src/test/java/io/spine/server/aggregate/AggregateRepositoryTest.java

```diff
@@ -0,0 +1,583 @@
+import io.spine.server.aggregate.given.AggregateRepositoryTestEnv.FailingAggregateRepository;
+import io.spine.server.aggregate.given.AggregateRepositoryTestEnv.GivenAggregate;
+import io.spine.server.aggregate.given.AggregateRepositoryTestEnv.ProjectAggregate;
+import io.spine.server.aggregate.given.AggregateRepositoryTestEnv.ProjectAggregateRepository;
+import io.spine.server.aggregate.given.AggregateRepositoryTestEnv.ReactingAggregate;
+import io.spine.server.aggregate.given.AggregateRepositoryTestEnv.ReactingRepository;
+import io.spine.server.aggregate.given.AggregateRepositoryTestEnv.RejectingRepository;
+import io.spine.server.aggregate.given.AggregateRepositoryTestEnv.RejectionReactingAggregate;
+import io.spine.server.aggregate.given.AggregateRepositoryTestEnv.RejectionReactingRepository;
+import io.spine.server.command.TestEventFactory;
+import io.spine.server.commandbus.CommandBus;
+import io.spine.server.model.HandlerMethodFailedException;
+import io.spine.server.model.Model;
+import io.spine.server.model.ModelTests;
+import io.spine.server.tenant.TenantAwareOperation;
+import io.spine.test.aggregate.ProjectId;
+import io.spine.test.aggregate.command.AggCreateProjectWithChildren;
+import io.spine.test.aggregate.command.AggStartProjectWithChildren;
+import io.spine.test.aggregate.event.AggProjectArchived;
+import io.spine.test.aggregate.event.AggProjectDeleted;
+import io.spine.testdata.Sample;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+import java.util.Iterator;
+import java.util.Set;
+
+import static io.spine.core.given.GivenTenantId.newUuid;
+import static io.spine.server.aggregate.AggregateRepository.DEFAULT_SNAPSHOT_TRIGGER;
+import static io.spine.validate.Validate.isNotDefault;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+/**
+ * @author Alexander Yevsyukov
+ */
+@SuppressWarnings({"InnerClassMayBeStatic", "ClassCanBeStatic"
+        /* JUnit 5 Nested classes cannot to be static. */,
+        "DuplicateStringLiteralInspection" /* Common test display names */})
+@DisplayName("AggregateRepository should")
+public class AggregateRepositoryTest {
+
+    private final TestActorRequestFactory requestFactory =
+            TestActorRequestFactory.newInstance(getClass());
+
+    private BoundedContext boundedContext;
+    private AggregateRepository<ProjectId, ProjectAggregate> repository;
+
+    private static ProjectId givenAggregateId(String id) {
```

Comment:
Can we please move the bounded context, the repository, the request factory + `given...` methods to the `...TestEnv` class? Then expose those via the test env API (static methods, e.g. `boundedContext()`, `given...` etc).